blockscout.star

```diff
@@ -0,0 +1,189 @@
+            },
+        ),
+    )
+
+    bs_visualize_service = plan.add_service(
+        name="blockscout-visualize" + args["deployment_suffix"],
+        config=ServiceConfig(
+            image=args["blockscout_visualizer_image"],
+            ports={
+                "blockscout": PortSpec(8050, application_protocol="http"),
+            },
+        ),
+    )
+
+    bs_frontend_service = plan.add_service(
+        name="blockscout-fe" + args["deployment_suffix"],
+        config=ServiceConfig(
+            image=args["blockscout_fe_image"],
+            ports={
+                "blockscout": PortSpec(
+                    args["blockscout_public_port"],
+                    application_protocol="http",
+                    wait="30s",
+                ),
+            },
+            public_ports={
+                "blockscout": PortSpec(
+                    args["blockscout_public_port"],
+                    application_protocol="http",
+                    wait="30s",
+                ),
+            },
+            env_vars={
+                "PORT": str(args["blockscout_public_port"]),
+                "NEXT_PUBLIC_NETWORK_NAME": "CDK zkEVM",
+                "NEXT_PUBLIC_NETWORK_ID": str(args["zkevm_rollup_chain_id"]),
+                "NEXT_PUBLIC_API_HOST": bs_backend_service.ip_address,
+                "NEXT_PUBLIC_API_PORT": str(
+                    bs_backend_service.ports["blockscout"].number
+                ),
+                "NEXT_PUBLIC_API_PROTOCOL": "http",
+                "NEXT_PUBLIC_API_WEBSOCKET_PROTOCOL": "ws",
+                "NEXT_PUBLIC_STATS_API_HOST": "http://{}:{}".format(
+                    bs_stats_service.ip_address,
+                    bs_stats_service.ports["blockscout"].number,
+                ),
+                "NEXT_PUBLIC_VISUALIZE_API_HOST": "http://{}:{}".format(
+                    bs_visualize_service.ip_address,
+                    bs_visualize_service.ports["blockscout"].number,
+                ),
+                "NEXT_PUBLIC_APP_PROTOCOL": "http",
+                "NEXT_PUBLIC_APP_HOST": "127.0.0.1",
+                "NEXT_PUBLIC_APP_PORT": str(args["blockscout_public_port"]),
+                "NEXT_PUBLIC_USE_NEXT_JS_PROXY": "true",
+                "NEXT_PUBLIC_AD_BANNER_PROVIDER": "none",
+                "NEXT_PUBLIC_AD_TEXT_PROVIDER": "none",
+            },
+        ),
+    )
+
+    return (bs_backend_service, bs_frontend_service)
+
+
+def run(plan, args):
+    rpc_url = None
+    ws_url = None
+    for service in plan.get_services():
+        if service.name == "zkevm-node-rpc" + args["deployment_suffix"]:
+            rpc_url = "http://{}:{}".format(
+                service.ip_address, service.ports["http-rpc"].number
+            )
+            ws_url = "ws://{}:{}".format(
+                service.ip_address, service.ports["ws-rpc"].number
+            )
+            break
+
+    if not (rpc_url and ws_url):
+        fail("Could not find the zkevm-node-rpc service")
```

Comment:
I really like this pattern, I feel like we could re-use this in `main.star` where services in step `x'` rely on services spin up in step `x`.